### PR TITLE
#5 eslint and prettier setup

### PR DESCRIPTION
**Pull request by eLeontev**
Issue: #5 

Changes:
- add eslint
- add prettier
- add scripts to run lint/prettier
- integrate eslint and prettier
- update all ts(x) files to fix all found eslint issues (mostly missed returned function type + React is not defined in tsx file)

### CODE REVIEW

**Comment by kishankr7979** [comment]
@eLeontev Please use husky and add a pre-commit file. Eslint should auto run whenever anyone try to commit the code via git.

**Comment by ps-89** [comment]
@eLeontev Can you please also check which new dependency under the hood uses the vulnerable package( @humanwhocodes/config-array@0.10.7), and if we can upgrade that dependency to the latest version.

Dependency anlaysis present at https://github.com/game1n/t8-cms/actions/runs/3164895830/jobs/5153533428

**Comment by ps-89** [comment]
@eLeontev We can squash all commits into one (or keep separate commits if there's a need for that), having 2 commits with the same message is a bit confusing at a later point in time when going through the git history.

How to squash last N commits - https://www.internalpointers.com/post/squash-commits-into-one-git

If this is something you don't know yet or find difficult, you can skip this, but this is quite helpful to learn.

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm run format
```
Comment (kishankr7979) [inline .husky/pre-commit:4]: use yarn

**Comment by eLeontev** [comment]
> @eLeontev We can squash all commits into one (or keep separate commits if there's a need for that), having 2 commits with the same message is a bit confusing at a later point in time when going through the git history.
> 
> How to squash last N commits - https://www.internalpointers.com/post/squash-commits-into-one-git
> 
> If this is something you don't know yet or find difficult, you can skip this, but this is quite helpful to learn.

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "singleQuote": true
```
Comment (kishankr7979) [inline .prettierrc.json:2]: add tabWidth, so that this should change on different IDEs/Code editors.

**Review by kishankr7979 — CHANGES_REQUESTED** [review]

**Comment by eLeontev** [comment]
> @eLeontev Can you also please check which new dependency under the hood uses the vulnerable package( @humanwhocodes/config-array@0.10.7), and if we can upgrade that dependency to the latest version(if possible)?
> 
> Dependency analysis present at https://github.com/game1n/t8-cms/actions/runs/3164895830/jobs/5153533428

I guess it's out of the MR, I just removed package-lock.json since it was not used in the project before. btw the audit system highlights that it's issue from react-scripts package and cannot be fixed until this package not updated yet.

Screenshot 2022-10-01 at 16 36 42

File: .husky/pre-commit (hunk 1 above)
Comment (eLeontev) [reply .husky/pre-commit:4]: looks like it does not support it https://yarnpkg.com/package/husky - recommendation is `npm`

File: .prettierrc.json (hunk 2 above)
Comment (eLeontev) [reply .prettierrc.json]: added with value `2` since it's the pattern used at the project

**Comment by eLeontev** [comment]
> @eLeontev We can squash all commits into one (or keep separate commits if there's a need for that), having 2 commits with the same message is a bit confusing at a later point in time when going through the git history.
> 
> How to squash last N commits - https://www.internalpointers.com/post/squash-commits-into-one-git
> 
> If this is something you don't know yet or find difficult, you can skip this, but this is quite helpful to learn.

ususally it's up to repo owners to set it up on PR merge, btw I squashed the commits

**Comment by ps-89** [comment]
@eLeontev Yes, we can squash while merging too, though if someone prefers to merge branches to the main branch via terminal, having a single commit with all the changes or having multiple commits with good messages helps.

Thanks a lot for your contribution, and welcome again to the project. I'm approving the PR and will wait for @kishankr7979 for any other comments and final approval. Looking forward to more contributions from you, as we've just started working on the project, so there'll be some good tasks that'll come up. Cheers!

**Review by ps-89 — APPROVED** [review]

**Review by kishankr7979 — APPROVED** [review]
LGTM :)

**Pull request merged**